src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/Binders/SimpleTypeModelBinder.cs

```diff
@@ -67,7 +84,7 @@ public Task BindModelAsync(ModelBindingContext bindingContext)
             {
                 bindingContext.ModelState.TryAddModelError(
                     bindingContext.ModelName,
-                    exception,
+                    exception.InnerException ?? exception,
```

Comment:
I'd expect this block to look very much like the existing code in https://github.com/aspnet/Mvc/blob/dev/src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/ModelBindingHelper.cs#L1066

For instance we should special case format exception, and it should include the comments about the fact that type converter wraps formatexceptions.

This code is really **replacing** the other code on the common execution path, so I expect it to have all the same bells and whistles.